ecmascript/parser/src/parser/typescript.rs

```diff
@@ -400,14 +405,19 @@ impl<'a, I: Tokens> Parser<'a, I> {
         debug_assert!(self.input.syntax().typescript());
 
         self.in_type().parse_with(|p| {
-            let start = cur_pos!();
+            let start = cur_pos!(); // todo: the starts below should not include the return_token
```

Comment:
I believe everything is good in this PR, but it shouldn't use this start for type predicates in functions. I believe this should go after the if statement below, but might be good to fix that in a separate PR.

Reply:
I agree with you.